.github/workflows/test.yaml

```diff
@@ -20,16 +33,60 @@ jobs:
     runs-on: 'ubuntu-latest'
     steps:
       - uses: holoviz-dev/holoviz_tasks/pre-commit@v0.1a17
+  setup:
+    name: Setup workflow
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ env.MATRIX }}
+    steps:
+      - name: Set matrix option
+        run: |
+          if [[ '${{ github.event_name }}' == 'workflow_dispatch' ]]; then
+            OPTION=${{ github.event.inputs.target }}
+          elif [[ '${{ github.event_name }}' == 'schedule' ]]; then
+            OPTION="full"
+          elif [[ '${{ github.event_name }}' == 'push' && '${{ github.ref_type }}' == 'tag' ]]; then
+            OPTION="full"
+          else
+            OPTION="default"
+          fi
+          echo "MATRIX_OPTION=$OPTION" >> $GITHUB_ENV
+      - name: Set test matrix with 'default' option
+        if: env.MATRIX_OPTION == 'default'
+        run: |
+          MATRIX=$(jq -nsc '{
+              "os": ["ubuntu-latest", "macos-latest", "windows-latest"],
+              "python-version": ["3.8", "3.11"],
+              "include": [
+                  {"os": "ubuntu-latest", "python-version": "3.9"},
+                  {"os": "ubuntu-latest", "python-version": "3.10"}
```

Comment:
I have added these two extra tests by default. You can remove it if you don't like it.

Reply:
Just wondering why you've added them?

Reply:
I added them to check how different versions of Python packages work. 

An example is that Python 3.8 only works up to Bokeh 3.1 and, therefore, not the latest Panel version. 

The next couple of releases for packages that follow [SPEC 0000](https://scientific-python.org/specs/spec-0000/) could drop Python 3.9 support.

Reply:
Ok I've removed them but it's good to know we can easily extend the matrix this way.